batcher/aligned-task-sender/src/commands.rs

```diff
@@ -0,0 +1,453 @@
+                        if e.kind() != ErrorKind::AlreadyExists {
+                            eprintln!("Error creating directory: {}", e);
+                            // Handle or log the error, but don't panic.
+                        }
+                    }
+
+                    Command::new("go")
+                        .arg("run")
+                        .arg(GROTH_16_PROOF_GENERATOR_FILE_PATH)
+                        .arg(format!("{:?}", i))
+                        .arg(dir_to_save_proofs)
+                        .status()
+                        .unwrap();
+                }
+            }
+        });
+        handles.push(handle);
+    }
+
+    for handle in handles {
+        let _ = handle.join();
+    }
+}
+
+pub async fn generate_and_fund_wallets(args: GenerateAndFundWalletsArgs) {
+    if matches!(args.network.into(), Network::Devnet) {
+        let Ok(eth_rpc_provider) = Provider::<Http>::try_from(args.eth_rpc_url.clone()) else {
+            error!("Could not connect to eth rpc");
+            return;
+        };
+        let Ok(chain_id) = eth_rpc_provider.get_chainid().await else {
+            error!("Could not get chain id");
+            return;
+        };
+        let amount_to_deposit_to_aligned =
+            parse_ether(&args.amount_to_deposit_to_aligned).expect("Ether format should be: XX.XX");
+
+        let file = match File::open(&args.private_keys_filepath) {
+            Ok(f) => f,
+            Err(err) => {
+                error!("Could not open private keys file: {}", err.to_string());
+                return;
+            }
+        };
+        let file_reader = BufReader::new(file);
+
+        let mut last_handle = None;
+        for (i, line) in file_reader.lines().enumerate() {
+            // Load the wallet
+            let private_key_str = line.unwrap();
+            let wallet = Wallet::from_str(private_key_str.trim())
+                .expect("Invalid private key")
+                .with_chain_id(chain_id.as_u64());
+
+            // Send funds to aligned from the wallet
+            let funded_wallet_signer =
+                SignerMiddleware::new(eth_rpc_provider.clone(), wallet.clone());
+            tokio::time::sleep(Duration::from_millis(50)).await; // To avoid overloading the RPC
+            last_handle = Some(tokio::spawn(async move {
+                if let Err(err) = deposit_to_aligned(
+                    amount_to_deposit_to_aligned,
+                    funded_wallet_signer.clone(),
+                    args.network.into(),
+                )
+                .await
+                {
+                    error!("Could not deposit to aligned, err: {:?}", err);
+                    return;
+                }
+                info!("Successfully deposited to aligned for wallet {}", i);
+            }));
+        }
+
+        // Wait for last tx to be sent
+        if let Some(handle) = last_handle {
```

Comment:
nit: This works, but we should await all handles, not only the last one

Reply:
done